crates/net/network/src/error.rs

```diff
@@ -214,7 +215,19 @@ impl SessionError for PendingSessionHandshakeError {
     fn is_fatal_protocol_error(&self) -> bool {
         match self {
             Self::Eth(eth) => eth.is_fatal_protocol_error(),
-            Self::Ecies(_) => true,
+            Self::Ecies(err) => matches!(
+                err.inner(),
+                ECIESErrorImpl::TagCheckDecryptFailed |
+                    ECIESErrorImpl::TagCheckHeaderFailed |
+                    ECIESErrorImpl::TagCheckBodyFailed |
+                    ECIESErrorImpl::InvalidAuthData |
+                    ECIESErrorImpl::InvalidAckData |
+                    ECIESErrorImpl::InvalidHeader |
+                    ECIESErrorImpl::Secp256k1(_) |
+                    ECIESErrorImpl::InvalidHandshake { .. } |
+                    ECIESErrorImpl::UnreadableStream |
+                    ECIESErrorImpl::StreamTimeout
+            ),
```

Comment:
could we convert this into a fn on `ECIESErrorImpl`?

Reply:
we only need this here, so this is fine